django/db/backends/sqlite3/schema.py

```diff
@@ -154,12 +154,20 @@ def is_self_referential(f):
             for index in model._meta.index_together
         ]
 
+        indexes = model._meta.indexes
+        if delete_fields:
+            indexes = [
+                index for index in indexes
+                if delete_fields[0].column not in index.fields
```

Comment:
Isn't this comparing column names (`.column`) to field names in `index.fields`? I think it should be `delete_fields[0].name` instead.